providers/gavc_download.rb

```diff
@@ -14,6 +17,9 @@ def whyrun_supported?
                                                  packaging: new_resource.packaging)
 
   resource_name = "Downloading file #{download_uri} specified by: #{new_resource.to_s}"
+  if new_resource.username && new_resource.password then
+    download_uri.sub!(/^(.*):\/\//, "\\1://#{new_resource.username}:#{new_resource.password}@")
```

Comment:
Could you please add tests to verify that the download URI has username and password values in it when appropriate?